coffee/lib/abstract-chosen.coffee

```diff
@@ -185,19 +185,21 @@ class AbstractChosen
           results += 1 if results_group.active_options is 0 and results_group.search_match
           results_group.active_options += 1
 
-        option.search_text = if option.group then option.label else option.html
+        text = if option.group then option.label else option.text
 
         unless option.group and not @group_search
-          search_match = this.search_string_match(option.search_text, regex)
+          search_match = this.search_string_match(text, regex)
           option.search_match = search_match?
 
           results += 1 if option.search_match and not option.group
 
           if option.search_match
-            if searchText.length
+            if query.length
               startpos = search_match.index
-              text = option.search_text.substr(0, startpos + searchText.length) + '</em>' + option.search_text.substr(startpos + searchText.length)
-              option.search_text = text.substr(0, startpos) + '<em>' + text.substr(startpos)
+              prefix = text.slice(0, startpos)
+              fix    = text.slice(startpos, startpos + query.length)
+              suffix = text.slice(startpos + query.length)
+              option.highlighted_html = "#{@escape_html(prefix)}<em>#{@escape_html(fix)}</em>#{@escape_html(suffix)}"
```

Comment:
For functions we use `this` instead of `@`, so please use `this.escape_html(prefix)`